client.go

```diff
@@ -501,10 +505,10 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 
 	if logger != nil {
 		switch v := logger.(type) {
-		case Logger:
```

Comment:
I'm not sure that this was an accident. It might have been prioritized to keep behavior the same for backwards compat. @evanphx would know more.

Reply:
Looks like he just wanted to prioritize the LevelLogger over the generic one. That seems like a good change since LevelLogger is more specific.

Reply:
Sounds good then.